silk-core/src/main/kotlin/net/silkmc/silk/core/event/ServerEvents.kt

```diff
@@ -0,0 +1,36 @@
+package net.silkmc.silk.core.event
+
+import net.minecraft.server.MinecraftServer
+import net.silkmc.silk.core.annotations.ExperimentalSilkApi
+
+/**
+ * Events related to the [MinecraftServer].
+ */
+@Suppress("unused") // receiver is for namespacing only
+val Events.server get() = ServerEvents
+
+@ExperimentalSilkApi
+object ServerEvents {
+
+    class ServerEvent(val server: MinecraftServer)
+
+    /**
+     * Called before the server is initialized.
+     */
+    val preStart = Event.bothImmutable<ServerEvent>()
```

Comment:
A also think that `starting` looks cleaner, but `preStart` immediately tells the user what exactly it does, so that was more important for me personally.

Reply:
Makes sense, `starting` kinda sounds like it's async and the server is starting in parallel